### PR TITLE
Adjust serial number in feed handler during replay to ensure

### PR DESCRIPTION
**Pull request by toregge**
proper semantics when using IGetSerialNum interface.

@baldersheim : please review

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]
This looks good to me, but I think that @geirst should have a look too.

File: searchcore/src/vespa/searchcore/proton/server/feedhandler.h [hunk 1]
```diff
@@ -73,9 +75,11 @@ class FeedHandler: private search::transactionlog::client::Callback,
     std::unique_ptr<TlsWriter>             _tlsMgrWriter;
     TlsWriter                             *_tlsWriter;
     TlsReplayProgress::UP                  _tlsReplayProgress;
-    // the serial num of the last message in the transaction log
+    // the serial num of the last feed operation processed by feed handler.
     SerialNum                              _serialNum;
     SerialNum                              _prunedSerialNum;
+    // the serial num of the last feed operation in the transaction log at startup before replay
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/feedhandler.h:81]: Consider adding a comment to __prunedSerialNum_ as well.

File: searchcore/src/vespa/searchcore/proton/server/i_inc_serial_num.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace proton {
+
+/**
+ * A simple interface for increasing a serial number and getting the new
+ * serial number.
+ **/
+class IIncSerialNum {
+public:
+    virtual ~IIncSerialNum() = default;
+    virtual search::SerialNum incSerialNum() = 0;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/i_inc_serial_num.h:16]: Consider using snake case.

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/vespa/searchcore/proton/server/feedhandler.h (hunk 1 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/server/feedhandler.h:82]: Done.

File: searchcore/src/vespa/searchcore/proton/server/i_inc_serial_num.h (hunk 2 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/server/i_inc_serial_num.h]: Fixed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**